test/sql/upsert.sql

```diff
@@ -0,0 +1,24 @@
+CREATE TABLE upsert_test(time timestamp PRIMARY KEY, temp float, color text);
+SELECT create_hypertable('upsert_test', 'time');
+INSERT INTO upsert_test VALUES ('2017-01-20T09:00:01', 22.5, 'yellow') RETURNING *;
+INSERT INTO upsert_test VALUES ('2017-01-20T09:00:01', 23.8, 'yellow') ON CONFLICT (time)
+DO UPDATE SET temp = 23.8 RETURNING *;
+INSERT INTO upsert_test VALUES ('2017-01-20T09:00:01', 78.4, 'yellow') ON CONFLICT DO NOTHING;
+SELECT * FROM upsert_test;
+
+-- Referencing constraints by name does not yet work on Hypertables. Check for proper error message.
+\set ON_ERROR_STOP 0
+INSERT INTO upsert_test VALUES ('2017-01-20T09:00:01', 12.3, 'yellow') ON CONFLICT ON CONSTRAINT upsert_test_pkey
+DO UPDATE SET temp = 12.3 RETURNING time, temp, color;
+\set ON_ERROR_STOP 1
+
+-- Test with UNIQUE index on multiple columns instead of PRIMARY KEY constraint
+CREATE TABLE upsert_test_unique(time timestamp, temp float, color text);
+SELECT create_hypertable('upsert_test_unique', 'time');
+CREATE UNIQUE INDEX color_idx ON upsert_test_unique (time, color);
+INSERT INTO upsert_test_unique VALUES ('2017-01-20T09:00:01', 22.5, 'yellow') RETURNING *;
+INSERT INTO upsert_test_unique VALUES ('2017-01-20T09:00:01', 21.2, 'brown');
+SELECT * FROM upsert_test_unique ORDER BY time, color DESC;
+INSERT INTO upsert_test_unique VALUES ('2017-01-20T09:00:01', 31.8, 'yellow') ON CONFLICT (time, color)
```

Comment:
I would like to see a test that shown picking the right arbiter index. Here there is only one. I'd like to see two different ones (say time and time,color) and being able to use either one correctly.